### PR TITLE
[FIX] In CharBPETokenizer, when Vocab or merges is None, unk_token cannot be used.

### PR DESCRIPTION
**Pull request by SeongBeomLEE**
Same Error #1120

Thanks.

## before
char_level_bep.py
```
if vocab is not None and merges is not None:
    tokenizer = Tokenizer(
        BPE(
            vocab,
            merges,
            dropout=dropout,
            unk_token=str(unk_token),
            end_of_word_suffix=suffix,
        )
    )
else:
    tokenizer = Tokenizer(BPE())
```

main.py
```
from tokenizers import CharBPETokenizer

tokenizer = CharBPETokenizer(
    unk_token = "[UNK]",
    suffix = "</w>",
)

tokenizer.train(
    files = './vocab.txt',
    vocab_size = 1000,
    min_frequency = 1,
    special_tokens = ["[PAD]", "[BOS]", "[EOS]", "[UNK]", "[SEP]", "[CLS]", "[MASK]"],
)

line = "나는 😀 😃 😄 축구를 😀 😃 😄 좋아한다."
pieces = tokenizer.encode(line)
print("CharBPETokenizer", pieces.tokens, tokenizer.decode(pieces.ids))

output:

CharBPETokenizer ['나는</w>', '축구를</w>', '좋아한다</w>', '.</w>'] 나는 축구를 좋아한다 .
```
## after

char_level_bep.py
```
if vocab is not None and merges is not None:
    tokenizer = Tokenizer(
        BPE(
            vocab,
            merges,
            dropout=dropout,
            unk_token=str(unk_token),
            end_of_word_suffix=suffix,
        )
    )
else:
    tokenizer = Tokenizer(BPE(unk_token=str(unk_token),))
```

main.py
```
from tokenizers import CharBPETokenizer

tokenizer = CharBPETokenizer(
    unk_token = "[UNK]",
    suffix = "</w>",
)

tokenizer.train(
    files = './vocab.txt',
    vocab_size = 1000,
    min_frequency = 1,
    special_tokens = ["[PAD]", "[BOS]", "[EOS]", "[UNK]", "[SEP]", "[CLS]", "[MASK]"],
)

line = "나는 😀 😃 😄 축구를 😀 😃 😄 좋아한다."
pieces = tokenizer.encode(line)
print("CharBPETokenizer", pieces.tokens, tokenizer.decode(pieces.ids))

output:

CharBPETokenizer ['나는</w>', '[UNK]', '[UNK]', '[UNK]', '축구를</w>', '[UNK]', '[UNK]', '[UNK]', '좋아한다</w>', '.</w>'] 나는 축구를 좋아한다 .
```

### CODE REVIEW

File: bindings/python/py_src/tokenizers/implementations/char_level_bpe.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def __init__(
                 )
             )
         else:
-            tokenizer = Tokenizer(BPE())
+            tokenizer = Tokenizer(BPE(unk_token=str(unk_token),))
```
Comment (Narsil) [inline bindings/python/py_src/tokenizers/implementations/char_level_bpe.py:48]:
```suggestion
            tokenizer = Tokenizer(BPE(unk_token=str(unk_token), dropout=dropout, end_of_word_suffix=suffx))
```

no ?

Comment (SeongBeomLEE) [reply bindings/python/py_src/tokenizers/implementations/char_level_bpe.py]: Yes 👍

File: bindings/python/py_src/tokenizers/implementations/char_level_bpe.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def __init__(
                 )
             )
         else:
-            tokenizer = Tokenizer(BPE())
+            tokenizer = Tokenizer(BPE(unk_token=str(unk_token), dropout=dropout, end_of_word_suffix=suffx))
```
Comment (Narsil) [inline bindings/python/py_src/tokenizers/implementations/char_level_bpe.py:48]:
```suggestion
            tokenizer = Tokenizer(BPE(unk_token=str(unk_token), dropout=dropout, end_of_word_suffix=suffix))
```

**Review by Narsil — APPROVED** [review]
LGTM. Thank you for this

**Pull request merged**